BackendIntegrationTests/BackendIntegrationTests.swift

```diff
@@ -215,23 +215,69 @@ class BackendIntegrationSK1Tests: XCTestCase {
 
         expect(customerInfo.entitlements.all.count) == 1
         let entitlements = self.purchasesDelegate.customerInfo?.entitlements
-        expect(entitlements?["premium"]?.isActive) == true
+        expect(entitlements?[Self.entitlementIdentifier]?.isActive) == true
 
         let anonUserID = Purchases.shared.appUserID
         let identifiedUserID = "\(#function)_\(anonUserID)_".replacingOccurrences(of: "RCAnonymous", with: "")
 
         let (identifiedCustomerInfo, created) = try await Purchases.shared.logIn(identifiedUserID)
         expect(created) == true
-        expect(identifiedCustomerInfo.entitlements["premium"]?.isActive) == true
+        expect(identifiedCustomerInfo.entitlements[Self.entitlementIdentifier]?.isActive) == true
 
         eligibility = await Purchases.shared.checkTrialOrIntroDiscountEligibility([productID])
         expect(eligibility[productID]?.status) == .ineligible
     }
 
+    func testUserHasNoEligibleOffersByDefault() async throws {
+        let (_, created) = try await Purchases.shared.logIn(UUID().uuidString)
+        expect(created) == true
+
+        let offerings = try await Purchases.shared.offerings()
+        let product = try XCTUnwrap(offerings.current?.monthly?.storeProduct)
+
+        expect(product.discounts).to(haveCount(1))
+        expect(product.discounts.first?.offerIdentifier) == "com.revenuecat.monthly_4.99.1_free_week"
+
+        let offers = await product.getEligiblePromotionalOffers()
+        expect(offers).to(beEmpty())
+    }
+
+    func testPurchaseWithDiscount() async throws {
+        let (_, created) = try await Purchases.shared.logIn(UUID().uuidString)
+        expect(created) == true
+
+        let offerings = try await Purchases.shared.offerings()
+        let product = try XCTUnwrap(offerings.current?.monthly?.storeProduct)
+
+        try await self.purchaseMonthlyOffering()
+        try self.verifyEntitlementWentThrough()
+
+        let offers = await product.getEligiblePromotionalOffers()
+        expect(offers).to(haveCount(1))
+        let offer = try XCTUnwrap(offers.first)
+
+        // TODO: ?
+        Thread.sleep(forTimeInterval: 5)
+
+        let result = try await Purchases.shared.purchase(product: product, promotionalOffer: offer)
+        let customerInfo = result.customerInfo
+
+        expect(customerInfo.entitlements.active).to(beEmpty())
+        expect(customerInfo.entitlements.all).to(haveCount(1))
+
+        let entitlement = try XCTUnwrap(customerInfo.entitlements[Self.entitlementIdentifier])
+
+        expect(entitlement.isActive) == false
+        expect(entitlement.willRenew) == true
+        expect(entitlement.periodType) == .intro
```

Comment:
This is working now, just gotta confirm it's the right behavior. I'm confused why the entitlement would be inactive?